### PR TITLE
Add clarifications and TOC to INSTALL.md

### PR DESCRIPTION
**Pull request by PaulMorris**
This fixes issue #472 INSTALL.md improvements based on Paul Morris' notes

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -7,6 +7,28 @@
...
 
+**Contents**
+
+- **[Background and Current Deployment Status](#background-and-current-deployment-status)**
+- **[Requirements Overview](#requirements-overview)**
+    - [Hardware](#hardware)
+    - [Software](#software)
+    - [Database](#database)
+    - [Mail](#mail)
+- **[Installation Instructions](installation-instructions)**
```
Comment (brainwane) [inline INSTALL.md]: This hyperlink doesn't seem to work. @jasonaowen saw a similar issue and fixed it in cdfff56c8bee5797bbdf042201505c386cbba0ec .

File: INSTALL.md [hunk 2]
```diff
@@ -7,6 +7,28 @@
...
+    - [Install prerequisites](#install-prerequisites)
+    - [Configure WildFly](#configure-wildFly)
```
Comment (brainwane) [inline INSTALL.md]: This hyperlink does not work for me.

File: INSTALL.md [hunk 3]
```diff
@@ -7,6 +7,28 @@
...
+    - [Configure Mail](#configure-mail)
+    - [Configure Database](#configure-database)
+    - [Build and deploy the application](#build-and-deploy-the-application)
+    - [Build documentation](#build-documentation)
+    - [Run automated tests](#run-automated-tests)
+    - [Maintain and update environment](#maintain-and-update-environment)
+- **[Production Deployment](production-deployment)**
```
Comment (brainwane) [inline INSTALL.md]: This hyperlink does not work for me.

File: INSTALL.md [hunk 4]
```diff
@@ -56,7 +76,9 @@ time, and will evolve as we understand it more.
 - **Storage**: 10 GB of storage for WildFly, the PSM repository, and its
   dependencies should be plenty.
 
-### Software
+## Software
+
+This is just an overview, see installation instructions below.
```
Comment (brainwane) [inline INSTALL.md]: This should be a semicolon, or two sentences, instead of a comma in one sentence.

File: INSTALL.md [hunk 5]
```diff
@@ -232,12 +254,13 @@ If you are using a production mail server, add a mail session with a JNDI name
 of `java:/Mail` to your application server with the appropriate credentials
 using the command line or web interface.
 
-### Database
+## Configure Database
```
Comment (brainwane) [inline INSTALL.md]: Nice fix for parallelism! :)

File: INSTALL.md [hunk 6]
```diff
@@ -320,12 +342,15 @@ EOF
...
-   your management console username and password, and do the
-   following: "Deployments > Add > Upload a new deployment > browse to
-   file." Alternatively, you can use the command line interface,
-   replacing the relevant paths below:
+1. Deploy the built app.  You can use the WildFly Management Console UI
+   at [http://localhost:9990/](http://localhost:9990/).  Log in with
+   your management console username and password (if you are not already logged
+   in).  Then do the following: "Deployments [tab] > Add [button] > Upload a new
+   deployment > browse to file."  And browse to this file:
+   {/path/to/psm}/psm-app/cms-portal-services/build/libs/cms-portal-services.ear
```
Comment (brainwane) [inline INSTALL.md]: Please surround the filename with tick marks to ensure it's styled as fixed-width code-style text.

> ` {/path/to/psm}/psm-app/cms-portal-services/build/libs/cms-portal-services.ear`

**Comment by brainwane** [comment]
Thanks for writing a descriptive set of commit messages. Could you please change "Some clarifications in INSTALL.md" to "Clarify instructions in INSTALL.md" to fit better with our commit message style guide?

Thanks!

**Comment by PaulMorris** [comment]
@cecilia-donnelly So some more info:  The .pgpass file was not actually related to the problem I ran into when installing.  The problem was just that my db password was not being recognized.  (Not sure how it got off, possibly user error.)  @jasonaowen helped me resolve this by just doing a hard reset of the password along with some refresh/reboot steps with the database.

The .pgpass file just provides the convenience of not having to enter your db password over and over during development cycles.  So it might be worth including instructions for it, but not strictly needed, and it may be something those familiar with postgres already know about (in the grey area between how much to include / assume).  But I'm glad to add instructions for it if we think they should go in.

**Comment by brainwane** [comment]
Sorry I hadn't looked back and seen that this required re-review. Looks good to me now. I agree with @cecilia-donnelly -- go ahead and merge this as is, and then, if it fits with your current tasks, make a new PR adding `.pgpass` instructions.

**Comment by PaulMorris** [comment]
Thanks @brainwane , will do.

**Pull request merged**